nucleus/payara-modules/notification-core/src/main/java/fish/payara/nucleus/notification/NotifierHandler.java

```diff
@@ -61,34 +61,75 @@
     private static final Logger LOGGER = Logger.getLogger(NotifierHandler.class.getName());
 
     private final PayaraNotifier notifier;
+    private final PayaraNotifierConfiguration config;
     private final String notifierName;
 
     private final Queue<PayaraNotification> notificationQueue;
 
-    public NotifierHandler(ServiceHandle<PayaraNotifier> notifierHandle) {
+    public NotifierHandler(final ServiceHandle<PayaraNotifier> notifierHandle) {
         this(notifierHandle, null);
     }
 
-    public NotifierHandler(ServiceHandle<PayaraNotifier> notifierHandle, PayaraNotifierConfiguration config) {
+    public NotifierHandler(final ServiceHandle<PayaraNotifier> notifierHandle, final PayaraNotifierConfiguration config) {
         this.notifier = notifierHandle.getService();
         this.notifierName = getNotifierName(notifierHandle.getActiveDescriptor());
         this.notificationQueue = new ConcurrentLinkedQueue<>();
+        this.config = config;
+    }
+
+    protected PayaraNotifierConfiguration getConfig() {
+        return config;
     }
 
     protected String getName() {
         return notifierName;
     }
 
+    protected void reconfigure() {
+        // Get the current configuration
+        PayaraNotifierConfiguration currentConfig = null;
+        if (config != null) {
+            currentConfig = PayaraConfiguredNotifier.class.cast(notifier).getConfiguration();
+            if (currentConfig == null) {
+                currentConfig = config;
+            }
+        }
+
+        final boolean enabled = config != null && valueOf(config.getEnabled());
+        final boolean wasEnabled = config != null && valueOf(currentConfig.getEnabled());
+
+        if (!enabled) {
+            if (wasEnabled) {
+                // If the notifier isn't enabled but was before
+                destroy();
+            }
+        } else {
+            if (wasEnabled) {
+                // If the notifier is enabled and was before
+                destroy();
+                bootstrap();
+            } else {
+                // If the notifier is enabled and wasn't before
+                bootstrap();
+            }
+        }
+    }
+
     protected void destroy() {
         notifier.destroy();
     }
 
+    @SuppressWarnings("unchecked")
     protected void bootstrap() {
+        // Set the configuration before bootstrapping the notifier
+        if (config != null) {
+            PayaraConfiguredNotifier.class.cast(notifier).setConfiguration(config);
```

Comment:
This is the kind of _set_ that you will not see coming should this not behave as expected and you start looking. I wonder if there is a better way where the config and the notifier interact a bit more obvious with each other.

Reply:
I know what you mean, but I can't currently think of a better way. Any ideas?

Reply:
No, would need to checkout the codebase and browse around this a little to get a better understanding of the relations involved. But as long as we are aware of this I think we could also just give it some time and at some point in the future a batter way becomes apparent to us - then we just have to also do the change :)